### PR TITLE
Initial add of a config module.

### PR DESCRIPTION
**Pull request by fuzzball81**
Adding a configuration module with unit tests. The first set of
features are functions to enable searching for config files.

- Updated travis and tox to support python 3.4 - 3.7
- Updated the pipenv to have the correct locked versions of pytest to
  support python 3.4 and 3.5
- Updated setup.py to support modules.

Signed-off-by: Jason Joyce <fuzzball81@gmail.com>

### CODE REVIEW

File: atkinson/config/search.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+def expand_user(path):
+    """Expand ~ to /home/<username>"""
+    if '~' in path:
+        path = str.replace(path, '~', os.path.expanduser('~'))
+    return path
+
+
+def config_search_paths(override_list=None):
+    """Gererate a list of paths to search for config files"""
```
Comment (jguiditta) [inline atkinson/config/search.py]: Typo Gererate -> Generate

File: tests/config/test_search.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        expected = search.expand_user('~/my_config_dir/config.yml')
+        assert search.get_config_file(overrides='~/my_config_dir') == expected
+
+
+def test_default_override_not_found():
+    """Test defult from override list"""
+    with patch('atkinson.config.search._check_available') as aval_mock:
+        aval_mock.side_effect = [False, True]
+        expected = search.expand_user('~/.atkinson/config.yml')
+        assert search.get_config_file(overrides=['~/my_config_dir']) == expected
```
Comment (jguiditta) [inline tests/config/test_search.py]: What happens if you pass in two valid overrides?

File: atkinson/config/search.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+def _check_available(filename):   # pragma: no cover
+    """Check to see if the filename exists and is a file"""
+    return os.path.exists(filename) and os.path.isfile(filename)
+
+
+def expand_user(path):
+    """Expand ~ to /home/<username>"""
+    if '~' in path:
+        path = str.replace(path, '~', os.path.expanduser('~'))
```
Comment (jguiditta) [inline atkinson/config/search.py]: I thought path.expanduser could take any string and do what this function is trying to do.  Is there a case that doesn't work, or is this doing something different than it sounds like?

Comment (fuzzball81) [reply atkinson/config/search.py]: It appears that it does now, I will remove that function and update the code to use os.path.expanduser.

File: tests/config/test_search.py (hunk 2 above)
Comment (fuzzball81) [reply tests/config/test_search.py]: The first directory that contains the matching file will be taken. Let me rework that test to make it a little more clear. That test is for fall back to the defaults, if the overrides do not contain the file then look in our default list for it.

File: atkinson/config/search.py (hunk 1 above)
Comment (fuzzball81) [reply atkinson/config/search.py]: Oops, good catch.

File: Pipfile [hunk 4]
```diff
@@ -4,12 +4,7 @@ verify_ssl = true
 name = "pypi"
 
 [dev-packages]
-flake8 = "*"
-pylint = "*"
```
Comment (jguiditta) [inline Pipfile:8]: Just to be sure, did you intend to remove these two?  ^

**Review by jguiditta — COMMENTED** [review]
Minor question aside, lgtm! :+1:

File: Pipfile (hunk 4 above)
Comment (fuzzball81) [reply Pipfile:8]: Yes, those are tools that I like to have in my development environment that not every one else does. I did add a flake8 test to tox and travis to catch issues.

**Review by jguiditta — APPROVED** [review]

**Pull request merged**